src/main/java/com/marklogic/kafka/connect/sink/DefaultSinkRecordConverter.java

```diff
@@ -16,11 +15,16 @@
 public class DefaultSinkRecordConverter implements SinkRecordConverter {
 
 	private DocumentWriteOperationBuilder documentWriteOperationBuilder;
-
 	private Format format;
 	private String mimeType;
-
+	private Boolean addTopicIndicated = true ; 
+	private String  mlcollection; 
```

Comment:
I don't believe this is a safe implementation and thus it needs to be changed. Everything except "mlcollection" is set when the converter is initialized, but then "mlcollection" has the potential to change its value at runtime.

I believe the risk occurs when receiving messages from multiple topics, which the config in marklogic-sink.properties indicates is possible - "topics" has a comment of "Topics to consume from [comma separated list for multiple topics]". 

Thus, there's a race condition here, as the converter could be handling messages from different topics at the same time, and the value of "mlcollection" cannot be guaranteed to be correct for a given sink record.